stdlib/builtin.rbi

```diff
@@ -560,10 +560,60 @@ class Enumerator<'a, 'b>
                 | -> Enumerator<['a, Integer], 'b>
 end
 
+class Encoding
+end
+
 class Regexp
   def self.compile: (String, *any) -> Regexp
+                  | (Regexp) -> Regexp
   def self.escape: (String) -> String
+  def self.last_match: -> MatchData?
+                     | (Integer) -> String?
+  def self.quote: (String) -> String
+  def self.try_convert: (any) -> Regexp?
+  def self.union: (*String) -> Regexp
+                | (Array<String>) -> Regexp
+                | (*Regexp) -> Regexp
+                | (Array<Regexp>) -> Regexp
+
+  def initialize: (String, *any) -> any
+                  | (Regexp) -> any
+  def ===: (String) -> bool
+  def =~: (String) -> Integer?
+  def casefold?: -> bool
+  def encoding: -> Encoding
+  def fixed_encoding?: -> bool
+  def match: (String) -> MatchData?
+           | (String, Integer) -> MatchData?
+           | (String) { (MatchData) -> void } -> void
+           | (String, Integer) { (MatchData) -> void } -> void
+  def match?: (String) -> bool
+            | (String, Integer) -> bool
+  def named_captures: -> Hash<String, Array<Integer>>
+  def names: -> Array<String>
+  def options: -> Integer
   def source: -> String
+  def ~: -> Integer?
+end
+
+class MatchData
+  def []: (Integer | String | Symbol) -> String?
+        | (Integer, Integer) -> Array<String>
+        | (Range<Integer>) -> Array<String>
+  def begin: (Integer | String | Symbol) -> Integer
+  def captures: -> Array<String>
+  def end: (Integer | String | Symbol) -> Integer
+  def length: -> Integer
+  def named_captures: -> Hash<String, String?>
+  def names: -> Array<String>
+  def offset: (Integer | String | Symbol) -> [Integer, Integer]
+  def post_match: -> String
+  def pre_match: -> String
+  def regexp: -> Regexp
+  def size: -> Integer
+  def string: -> String
+  def to_a: -> Array<String>
+  def values_at: (*(Integer | String | Symbol)) -> Array<String>
```

Comment:
`... -> Array<String?>`

Reply:
Similar problem as https://github.com/soutaro/steep/pull/58#discussion_r223859627. 😓 

When passing `Integer`, it can return `nil`.

```
$ irb
irb(main):001:0> match = /(?<foo>a)/.match("a")
=> #<MatchData "a" foo:"a">
irb(main):002:0> match.values_at(0,1,2,:foo)
=> ["a", "a", nil, "a"]
```

I will fix it. 👍

Reply:
I fixed via 8f4db9b, but actually should we do this? 👇 

```
def values_at: (*(String | Symbol)) -> Array<String>
             | (*(String | Symbol | Integer)) -> Array<String?>